### ISSUE TITLE
BUG: _get_predict_end raises with time zones

### ISSUE THREAD

**Issue by antvig** [statsmodels/statsmodels#3912]
Hi, 
If the index of a series used to create a TimeSeriesModel is 'time zone aware', the following error occure : 

```
  File "C:\Users\SESA408771\Documents\VENVS\tsregcore_env\lib\site-packages\statsmodels\tsa\arima_model.py", line 734, in predict
    end, out_of_sample = self._get_predict_end(end, dynamic)
  File "C:\Users\SESA408771\Documents\VENVS\tsregcore_env\lib\site-packages\statsmodels\tsa\arima_model.py", line 675, in _get_predict_end
    return super(ARMA, self)._get_predict_end(end)
  File "C:\Users\SESA408771\Documents\VENVS\tsregcore_env\lib\site-packages\statsmodels\tsa\base\tsa_model.py", line 175, in _get_predict_end
    if dtend > self.data.dates[-1]:
  File "pandas\tslib.pyx", line 1106, in pandas.tslib._Timestamp.__richcmp__ (pandas\tslib.c:20503)
  File "pandas\tslib.pyx", line 1167, in pandas.tslib._Timestamp._assert_tzawareness_compat (pandas\tslib.c:21519)
TypeError: Cannot compare tz-naive and tz-aware timestamps
```

**Comment by josef-pkt** [comment]
This might be a "feature" and not a bug if end has a different type of time zone than the index used for the estimation data. That is, this might be pandas behavior that does the comparison.

Can you provide a full example or test case? 
This is the first time, AFAIR, that time zones show up in an issue.

**Comment by josef-pkt** [comment]
@ChadFulton ?  Do you know if there is anything time zone specific in statsmodels?

**Comment by antvig** [comment]
I use `end` and `start `parameters in the ARIMA predict function. But even if the end and start have the same time zone than the index of the series, the error is raised. 
The issue comes from `TimeSeriesModel._get_predict_end`  (`_get_predict_start`) : this method call ` end = _str_to_date(str(end))` which remove the timezone information... This datetime is later compared to the index of the series which still has the timezone information leading to `TypeError: Cannot compare tz-naive and tz-aware timestamps.`

I think `_str_to_date `method should be modified to handle time zone.

**Comment by ChadFulton** [comment]
The master version of Statsmodels includes a rewrite of the date / time handling code, and the function that's raising the exception (`TimeSeriesModel. _get_predict_end `) no longer exists.

If you happen to have a minimal example that reproduces the problem, it would be great to see if it still exists in the updated code.

Thanks for letting us know!

**Comment by josef-pkt** [comment]
This can be closed if we don't get an example to check whether it is still broken with current master (or 0.9)

**Comment by antvig** [comment]
Sorry for the late answer.

The example that reproduce the error :

[exemple_issue#3912.zip](https://github.com/statsmodels/statsmodels/files/1371142/exemple_issue.3912.zip)

**Comment by josef-pkt** [comment]
@antvig thanks for the example. 
It runs in current master without exception, and plotting the last 100 to 150 observations of observed and predicted looks like it's correct.

copying exmple here given that it is short and just one script file

---
```
from statsmodels.tsa.arima_process import arma_generate_sample
import statsmodels.api as sm
import pandas as pd 

p = 3 
q = 2 

# GENERATE (3,2) ARMA MODEL
array = arma_generate_sample(ar=[1,-0.1,-0.1,-0.1], ma=[1, 0.8,0.8], nsample=5000)

# CREATE A TIME ZONE AWARE INDEX
idx = pd.date_range(start=pd.datetime(2017,10,10), periods = 5000, freq='1S')
idx = idx.tz_localize('UTC')

# CREATE THE LEARNING DATA
learing_series = pd.Series(array, index=idx)

# LEARN THE ARMA MODEL
arma_model = sm.tsa.ARMA(learing_series, (p, q)).fit(trend = 'nc', transparams=True)

# CREATE A PREDICTION HORIZON (SAME TIME ZONE THAN idx)
prediction_horizon = idx[-1] + pd.Timedelta('60S')

# APPLY THE MODEL
arma_model.predict(end=prediction_horizon)
```
----

and quick plot
```
predicted = arma_model.predict(end=prediction_horizon)
import matplotlib.pyplot as plt
plt.figure(figsize=(8, 6))
plt.plot(learing_series.iloc[-100:])
plt.plot(predicted.iloc[-150:])
```

**Comment by josef-pkt** [comment]
closing, looks like bug was fixed for 0.9 with the date/time handling refactor
